pywinauto/hooks.py

```diff
@@ -0,0 +1,276 @@
+        self.handler = 0
+        self.pressed_keys = []
+        self.keyboard_id = None
+        self.mouse_id = None
+
+    def hook(self, keyboard=True, mouse=False):
+        """Hook mouse and/or keyboard events"""
+        if not mouse and not keyboard:
+            return
+
+        if keyboard:
+            def keyboard_low_level_handler(code, event_code, kb_data_ptr):
+                """Execute when keyboard low level event was catched"""
+                key_code = kb_data_ptr[0]
+                current_key = self.ID_TO_KEY[key_code]
+                event_type = self.event_types[event_code]
+
+                if event_type == 'key down':
+                    self.pressed_keys.append(current_key)
+
+                if event_type == 'key up':
+                    self.pressed_keys.remove(current_key)
+
+                event = KeyboardEvent(current_key, event_type, self.pressed_keys)
+
+                if self.handler != 0:
+                    self.handler(event);
+
+                return windll.user32.CallNextHookEx(self.keyboard_id, code, event_code, kb_data_ptr)
+
+            keyboard_pointer = _callback_pointer(keyboard_low_level_handler)
+
+            windll.kernel32.GetModuleHandleW.restype = wintypes.HMODULE
+            windll.kernel32.GetModuleHandleW.argtypes = [wintypes.LPCWSTR]
+            self.keyboard_id = windll.user32.SetWindowsHookExA(self.WH_KEYBOARD_LL, keyboard_pointer,
+                                                               windll.kernel32.GetModuleHandleA(None),
+                                                               0)
+
+        if mouse:
+            def mouse_low_level_handler(code, event_code, kb_data_ptr):
+                """Execute when mouse low level event was catched"""
+                current_key = self.MOUSE_ID_TO_KEY[event_code]
+                if current_key != 'Move':
+                    event_type = self.MOUSE_ID_TO_EVENT_TYPE[event_code]
+                    event = MouseEvent(current_key, event_type)
+
+                    if self.handler != 0:
+                        self.handler(event);
+
+                return windll.user32.CallNextHookEx(self.mouse_id, code, event_code, kb_data_ptr)
+
+            mouse_pointer = _callback_pointer(mouse_low_level_handler)
+            self.mouse_id = windll.user32.SetWindowsHookExA(self.WH_MOUSE_LL, mouse_pointer,
+                                                            windll.kernel32.GetModuleHandleA(None), 0)
+
+        self.listen()
+
+    def unhook_mouse(self):
+        """Unhook mouse events"""
+        windll.user32.UnhookWindowsHookEx(self.mouse_id)
+
+    def unhook_keyboard(self):
+        """Unhook keyboard events"""
+        windll.user32.UnhookWindowsHookEx(self.keyboard_id)
+
+    def listen(self):
+        """Listen events"""
+        atexit.register(windll.user32.UnhookWindowsHookEx, self.keyboard_id)
+        atexit.register(windll.user32.UnhookWindowsHookEx, self.mouse_id)
+        while True:
```

Comment:
@vasily-v-ryabov, I want to understand the use case of a such infinite loop. My understanding was a bit different. I supposed the hook should be activated at the beginning of a user script and help to interrupt the script when a predefined keys sequence has been pressed. This is a simple example I thought about
```python
hk = Hook()
hk.handler = fire_an_event_on_a_key_press('Ctrl-L') 
hk.hook(keyboard = True)

app = Application().start("gui app.exe")
dlg = app.Dialog
while True:
    dlg.do_an_extensieve_job...()
    if hook_event_occured:
        return
```
Using the hook will allow to stop the user script without bringing the Python console to foreground as it can be challenging when the script constantly interacts with GUI elements.